src/MoBi.Presentation/Presenter/SelectObjectInTreePresenter.cs

```diff
@@ -68,7 +75,43 @@ public bool IsValidSelection(ObjectBaseDTO selectedDTO)
 
       public void InitTreeStructure(IReadOnlyList<ObjectBaseDTO> entityDTOs)
       {
-         _view.BindTo(entityDTOs);
+         _view.Display(entityDTOs.Select(mapToNode).ToList());
+      }
+
+      private ITreeNode mapToNode(ObjectBaseDTO dto)
+      {
+         switch (dto)
+         {
+            case SpatialStructureDTO spatialStructureDTO:
+               return getSpatialStructureNode(spatialStructureDTO);
+            case BuildingBlockDTO buildingBlockDTO:
+               return getBuildingBlockNode(buildingBlockDTO);
+            default:
+               return _spatialStructureNodeMapper.MapFrom(dto);
+         }
+      }
+
+      private HierarchicalStructureNode getBuildingBlockNode(BuildingBlockDTO buildingBlockDTO)
+      {
+         var buildingBlockNode = _spatialStructureNodeMapper.MapFrom(buildingBlockDTO);
+         buildingBlockDTO.Builder.MapAllUsing(_spatialStructureNodeMapper).Each(buildingBlockNode.AddChild);
+         return buildingBlockNode;
+      }
+
+      private HierarchicalStructureNode getSpatialStructureNode(SpatialStructureDTO spatialStructureDTO)
+      {
+         var spatialStructureNode = _spatialStructureNodeMapper.MapFrom(spatialStructureDTO);
+
+         if (spatialStructureDTO.MoleculeProperties != null)
+            spatialStructureNode.AddChild(_spatialStructureNodeMapper.MapFrom(spatialStructureDTO.MoleculeProperties));
```

Comment:
that woudl be nice if we did nto have to do all this defensive programming all the time
Maping null returns null and does not cras and there is a way to Adda child and bypass if it's null. That would make code easier. Not for this PR but feels like we are doubting ourselves al lot here